### ISSUE TITLE
Clean up source dependency build artefacts

### ISSUE THREAD

**Issue by thumphries** [haskell-mafia/mafia#138]
I have 22GB worth of build artefacts inside `dist` dirs in source dependencies. Seems like this would be easy to nuke inside `mafia clean`, i.e. remove `lib/source-dep/dist` for each source dep.

Any reason not to implement this? @jystic

**Comment by erikd-ambiata** [comment]
`mafia clean` cleans the build artifacts from the current project (much like `cabal clean` does). I think making it also clean all the build artefacts for the required libraries would be a bad idea, especially for anything that uses amazonka.

However i would be :+1: on an extra command option like `mafia clean --purge`.

**Comment by jacobstanley** [comment]
I think if you delete those directories it will corrupt your cache? need to verify that.

Ideally we should treat source dependencies the same as hackage ones and build a copy of the source in the global cache, that's the real fix I think.

**Comment by thumphries** [comment]
> I think if you delete those directories it will corrupt your cache?

That's... concerning. Why? Data files?

**Comment by thumphries** [comment]
> I think making it also clean all the build artefacts for the required libraries would be a bad idea, especially for anything that uses amazonka.

AFAIK you don't get any reuse out of those `dist` dirs, since they get dropped into the global cache and then symlinked back.

**Comment by jacobstanley** [comment]
> That's... concerning. Why? Data files?

Yeah it is, I'm not 100% on that, I just have my suspicions that's it could be a problem
